Formula/ktor.rb

```diff
@@ -0,0 +1,18 @@
+class Ktor < Formula
+  desc "CLI tool that creates porjects with ktor framework setup"
+  homepage "https://github.com/ktorio/ktor-cli"
+  url "https://files.pkg.jetbrains.space/public/p/ktor-cli-tool/mac-release/ktor-0.0.1.tar.gz"
+  sha256 "21056136165fe68bb68672a0035d41f61b940889d1d661093f44577567bf217f"
+  license "Apache-2.0"
+
+  uses_from_macos "curl"
+  uses_from_macos "unzip"
+
+  def install
+    bin.install "ktor"
+  end
+
+  test do
+    system "ktor", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the review!
Regarding the test, this program generates new Ktor projects as well as it installs JDK and Gradle in case it was not found on the machine. In order to do this, it creates a `~/.ktor` directory with `ktor.rc` file inside as well as it downloads the JDK. After that, it creates a new Ktor project.

This behaviour fails on your CI because apparently the app is not permitted to create `~/.ktor` directory.
Please note that we have our own CI (https://ktor.teamcity.com/project/Ktor_ProjectKtorCLI?mode=builds#all-projects) on TeamCity that has integration tests with docker and different system configurations.

Reply:
> as well as it installs JDK and Gradle in case it was not found on the machine.

It really shouldn't do that when it can just depend on them in the formula.

Reply:
Makes sense, but we need to create a `~/.ktor/ktor.rc` file anyway while it's not possible to achieve on your CI. Is it?